protobuf/lib/src/protobuf/coded_buffer_writer.dart

```diff
@@ -337,11 +337,22 @@ class CodedBufferWriter {
         _writeVarint32(value ? 1 : 0);
         break;
       case PbFieldType._BYTES_BIT:
-        _writeBytesNoTag(
-            value is Uint8List ? value : Uint8List.fromList(value));
+        final List<int> bytes = value;
+        if (bytes is Uint8List) {
+          _writeBytesNoTag(bytes);
+        } else if (bytes.isEmpty) {
+          writeInt32NoTag(0);
```

Comment:
Consider introducing a `_writeEmptyBytes()` helper and use it here and other places where we basically intend to write an empty byte array.

```
@pramga('vm:prefer-inline')
@...
void _writeEmptyBytes() {
  writeInt32NoTag(0);
}
```

Reply:
I added the helper but I didn't add the pragmas. We have a bunch of other one-liners in this file, I'd like to experiment with inlining them manually separately.